documentation/design-docs/data-catalog.md

```diff
@@ -0,0 +1,497 @@
+orchestration/monitoring tools it provides on top of the Core features
+are not high priority for us.
+As such, we think it would be  more prudent for us to build with dbt Core
+and design our own orchestration/monitoring/authentication integrations on top.
+Hence, when this doc refers to "dbt", we are actually referring to dbt Core.
+
+The downside of this choice is that we would have to choose a separate tool for
+orchestrating and monitoring our DAGs if we move forward with dbt. This is an
+important fact to note in our decision, because [orchestrators are notoriously
+controversial](https://stkbailey.substack.com/p/what-exactly-isnt-dbt):
+
+> Every other application, besides dbt, is an orchestrator. Every application
+> pays the price for it. Every person hates every application as soon as it
+> starts orchestrating. To orchestrate is to grapple with reality, to embrace
+> mortality, to welcome death.
+
+As such, we evaluate this choice with an eye towards options for third-party
+orchestration and monitoring.
+
+### Demo
+
+See the
+[`jeancochrane/dbt-spike`](https://github.com/ccao-data/data-architecture/tree/jeancochrane/dbt-spike/dbt)
+branch of this repo for an example of a simple dbt setup using our views.
+
+### Requirements matrix
+
+| Requirement | dbt | Notes |
+| ----------- | --- | ----- |
+| Referential integrity    | ✅ | Via tests. |
+| Schema consistency       | ✅ | Via tests. |
+| Documentation            | ✅ | Via docs generator. |
+| Orchestration/automation | ❌  | Requires dbt Cloud, or integration with another service. |
+| Simpler joins/views      | 🟡 | Can't simplify views, but can help factor them out and document them better. |
+| Data validation          | ✅ | Via tests. |
+| Automated flagging       | ❌  | Requires dbt Cloud, or integration with another service. |
+| Monitoring               | ❌  | Requires dbt Cloud, or integration with another service. |
+| Athena integration       | ✅ | Via `dbt-athena-community` plugin. |
+
+After some discussion, we consider the lack of orchestration, automation,
+automated flagging, and monitoring to be acceptable. The only mission critical
+pieces are are flagging and monitoring, and we provide some thoughts on how
+to accomplish those goals using external services in the [Design](#design)
+section below.
+
+### Design
+
+Here is a sketch of how we plan to implement a workflow for cataloging
+and validating our data using dbt:
+
+#### Code organization
+
+* Our dbt configuration will be stored in a `dbt/` folder in the
+  `data-architecture` repo.
+* View definitions will be stored in a `models/` subdirectory, with
+  nested subdirectories for each schema (e.g. `default/`, `location/`).
+* Tests will be defined in a `schema.yml` file, and docs will be defined
+  as doc blocks in a `docs.md` file.
+    * It may end up being wise to split these files up by schema.
+* We will have two profiles, one for `dev` and one for `prod`, which will point
```

Comment:
Keep in mind that each environment will need to fully rebuild the DAG, which could take hours if it includes the CTAS. One possibility might be to "steal" the DAG state from the main environment (via GH Actions cache), such that only the changed parts of the DAG need to be rebuilt in the test environment.